### PR TITLE
Cleaned up and reorder skip functions from e2e framework

### PR DESCRIPTION
**Pull request by alejandrox1**
This patch moves skip-related functions from framework/util.go to
another file.
It also reorders the functions here to keep this file a bit more organized and easier to parse through for people.

Signed-off-by: alejandrox1 <alarcj137@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This patch moves skip-related functions from framework/util.go to another file.

**Which issue(s) this PR fixes**:
tangentially related to #77095

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/area e2e-test-framework
/sig testing
/priority important-soon
/milestone v1.17

### CODE REVIEW

**Comment by alejandrox1** [comment]
/assign @oomichi @timothysc @andrewsykim 
ptal

File: test/e2e/framework/skip.go [hunk 1]
```diff
@@ -51,34 +93,13 @@ func SkipUnlessNodeCountIsAtMost(maxNodeCount int) {
 	}
 }
 
-// SkipUnlessAtLeast skips if the value is less than the minValue.
-func SkipUnlessAtLeast(value int, minValue int, message string) {
```
Comment (fejta) [inline test/e2e/framework/skip.go]: Can we move the code back here? Same with other functions that are getting moved around inside the same file.

File: test/e2e/framework/skip.go [hunk 2]
```diff
@@ -136,18 +150,50 @@ func SkipIfNodeOSDistroIs(unsupportedNodeOsDistros ...string) {
...
-func SkipIfContainerRuntimeIs(runtimes ...string) {
-	for _, containerRuntime := range runtimes {
-		if containerRuntime == TestContext.ContainerRuntime {
-			skipInternalf(1, "Not supported under container runtime %s", containerRuntime)
-		}
+// ServerVersionGTE returns true if v is greater than or equal to the server
+// version.
+//
+// TODO(18726): This should be incorporated into client.VersionInterface.
+func ServerVersionGTE(v *utilversion.Version, c discovery.ServerVersionInterface) (bool, error) {
```
Comment (oomichi) [inline test/e2e/framework/skip.go]: ServerVresionGTE is called only from SkipUnlessServerVersionGTE which is moved to skip.go together.
Can we make the function local as `serverVersionGTE`?

**Review by oomichi — COMMENTED** [review]
Overall good for me, just one small comment.

File: test/e2e/framework/skip.go (hunk 2 above)
Comment (alejandrox1) [reply test/e2e/framework/skip.go]: ha :sweat_smile: forgot to check this. thank you for the catch!
fixed

File: test/e2e/framework/skip.go (hunk 1 above)
Comment (alejandrox1) [reply test/e2e/framework/skip.go]: My intent is to try to organize this file, kind of group functions alphabetically (order lexicographically by Skip[If|Unless]*) to not have the same mess as in util.go

Comment (fejta) [reply test/e2e/framework/skip.go]: That's not what the PR says though.

Consider separating the "sort function order" and "move functions into a different file" changes into separate commits.

Comment (alejandrox1) [reply test/e2e/framework/skip.go]: I'm inclined to keep everything together as it is a very simple and minor change but if you really think this should be separated then will do so.
for the time being, i retitled this PR and updated the description.

**Comment by oomichi** [comment]
Thanks for updating,

/lgtm

**Review by fejta — APPROVED** [review]

**Review by fejta — COMMENTED** [review]
/lgtm

**Pull request merged**